### ISSUE TITLE
False Positive in RCS0005 (Add blank line before #endregion) when #endregion follows #endif / #error / #line / #nullable / #warning, plus a blank line

### ISSUE THREAD

**Issue by Bergam64** [dotnet/roslynator#1626]
**Product and Version Used**: 
VS extension **Roslynator 2022** v4.12.11

**Steps to Reproduce**:
- (Example with `#endif`. The issue is the same with `#error`, `#line`, `#nullable` or `#warning`)
- Enable [`RCS0005`](https://josefpihrt.github.io/docs/roslynator/analyzers/RCS0005) in your `.editorconfig` file (disabled by default).
- Open a C# file and follow example below.

**Actual Behavior**:
```csharp
    #region R

#if false
    // ...
#endif

    #endregion // <-- RCS0005 is wrongly reported here, even though there's a blank line above!
```

**Expected Behavior**:
```csharp
    #region R

#if false
    // ...
#endif

    #endregion // <-- No RCS0005 reported here.
```

**Note:** similar to [issue 1528](https://github.com/dotnet/roslynator/issues/1528).

**Comment by josefpihrt** [comment]
fixed with #1610
